app/models/account.rb

```diff
@@ -580,6 +580,13 @@ def generate_site_access_code
     self.site_access_code ||= SecureRandom.hex(5) if provider?
   end
 
+  def set_default_master
+    if self.provider? && self.master == nil
+      self.master = false
+    end
+    true
```

Comment:
Oh because of old callback way to check for `false` you have to return `true`
Good catch ;)